Week2/part1.js

```diff
@@ -0,0 +1,130 @@
+const util = require('util');
+const mysql = require('mysql');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'new_world'
+});
+
+// Promisify the bind function of query function of connection object
+// Pass connection object (because bind expects "this")
+// Afterwards execQuery will work as a function that returns a promise but
+// we don't have to call "then" over that promise
+const execQuery = util.promisify(connection.query.bind(connection));
+
+const countryLang1 = [
+  {
+    'countryCode': 'ABW',
+    'language': 'XXX',
+    'isOfficial': 'F',
+    'percentage': 3,
+  }
+];
+const countryLang2 = [
+  {
+    'countryCode': 'KEN',
+    'language': 'XXX',
+    'isOfficial': 'F',
+    'percentage': 13,
+  }
+];
+
+switch (process.argv[2]) {
+  case 'help':
+    console.log("Usage of the program:");
+    console.log("node part1.js help");
+    console.log("node part1.js 1 'TURKEY' (1. What is the capital of country X ? )");
+    console.log("node part1.js 2 'Middle East' (2. List all the languages spoken in the region Y )");
+    console.log("node part1.js 3 'Turkish' (3. Find the number of cities in which language Z is spoken )");
+    console.log("node part1.js 4 'Turkish' (4. Are there any countries that have A) Same official language B) Same region If yes, display those countries. If no, display TRUE or FALSE)");
+    console.log("node part1.js 5 (5. List all the continents with the number of languages spoken in each continent)");
+    console.log("node part1.js 6 'KEN' 'XXX' 'F' 13");
+    break;
+  case '1':
+    getCapitalOfCountry(process.argv[3]);
```

Comment:
In this case, you need to consider the async nature of the query. You might want to refactor your functions and use ```await``` and ```async```. I posted some great examples of that in slack. =)